### DISCUSSION TITLE
Handling OAuth Registration Errors for Unverified Accounts in Rodauth: Expected Behavior and Possible Solutions

### DISCUSSION THREAD

**Discussion by Halvanhelv** [janko/rodauth-omniauth#18] [Q&A] [answered]
Hello!

I have a question regarding the registration and verification process for users. I noticed that when a new user has registered but has not yet completed the verification process, and tries to register through OAuth with the same email, Rodauth displays the following error: "The account matching the external identity is currently awaiting verification." Is this behavior expected? And is there a possibility to change this process to avoid such errors during registration through external services?

Thank you for your help!

**Comment by janko** [comment, marked as answer]
Interesting, it might make more sense if rodauth-omniauth made the account verified in that case, because that's the case when the account is created through OmniAuth login. What do you think?

**Reply by Halvanhelv** [reply]
I think so, it would make more sense to just verify the account automatically anyway

**Reply by janko** [reply]
I agree, I will try to make that change sometime this week.

**Reply by janko** [reply]
I took me a while, but I just pushed https://github.com/janko/rodauth-omniauth/commit/159b9407c9eb0fc978506022faa0890dc11d62d2 to master, which I plan to release soon.
